controllers/btpoperator_controller_test.go

```diff
@@ -249,200 +248,314 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
+			It("resources with new names should be created, resources with old names should be deleted, unchanged resources should stay and receive new chart version", func() {
+				updateManifestsNum := 3
+				err := moveOrCopyNFilesFromDirToDir(updateManifestsNum, false, getApplyPath(), getTempPath())
+				Expect(err).To(BeNil())
+
+				oldObjs, err := manifestHandler.CollectObjectsFromDir(getTempPath())
+				Expect(err).To(BeNil())
+				oldUns, err := manifestHandler.ObjectsToUnstructured(oldObjs)
+				Expect(err).To(BeNil())
+
+				err = ymlutils.CopyManifestsFromYamlsIntoOneYaml(getTempPath(), getToDeleteYamlPath())
 				Expect(err).To(BeNil())
 
-				simulateRestart(ctx, cr)
+				err = ymlutils.AddSuffixToNameInManifests(getTempPath(), suffix)
+				Expect(err).To(BeNil())
 
-				Expect(pullFromBtpManagerConfigMap(oldChartVersionKey)).To(Equal(initChartVersion))
-				Expect(pullFromBtpManagerConfigMap(currentCharVersionKey)).To(Equal(initChartVersion))
+				err = moveOrCopyNFilesFromDirToDir(updateManifestsNum, true, getTempPath(), getApplyPath())
+				Expect(err).To(BeNil())
 
-				oldCount, newCount := countResources()
-				Expect(pullFromBtpManagerConfigMap(currentCharVersionKey)).To(Equal(pullFromBtpManagerConfigMap(oldChartVersionKey)))
-				Expect(newCount).To(BeEquivalentTo(oldCount))
-				Expect(oldCount >= minimalExpectedElementsCount).To(BeTrue())
+				err = ymlutils.UpdateChartVersion(chartUpdatePath, newChartVersion)
+				Expect(err).To(BeNil())
+
+				Eventually(reconciler.workqueueSize).WithTimeout(time.Second * 5).WithPolling(time.Millisecond * 100).Should(Equal(0))
+				_, err = reconciler.Reconcile(ctx, controllerruntime.Request{NamespacedName: apimachienerytypes.NamespacedName{
+					Namespace: cr.Namespace,
+					Name:      cr.Name,
+				}})
+				Expect(err).To(BeNil())
 
-				Eventually(updateCh).Should(Receive(matchReadyCondition(types.StateReady, metav1.ConditionTrue, UpdateCheckSucceeded)))
+				actualNumOfOldResources, err := countResourcesForGivenChartVer(gvks, initChartVersion)
+				Expect(err).To(BeNil())
+				Expect(actualNumOfOldResources).To(Equal(0))
+				actualNumOfNewResources, err := countResourcesForGivenChartVer(gvks, newChartVersion)
+				Expect(err).To(BeNil())
+				Expect(actualNumOfNewResources).To(Equal(initResourcesNum))
+				assertResourcesRemoval(oldUns...)
 			})
 		})
 
-		When("resources should stay as they are and we bump chart version", Label("test-update"), func() {
-			It("existing resources has new version set and we delete nothing (check if any resources with old labels exists -> should be 0)", func() {
-				err := ymlutils.UpdateVersion(updatePath, newChartVersion)
+		When("remove some manifests and bump chart version", Label("test-update"), func() {
+			It("resources without manifests should be removed, unchanged resources should stay and receive new chart version", func() {
+				allManifests := 99 // higher number than number of manifests
```

Comment:
Is it safe assumption that 99 is greater than any possible number of manifests?

Reply:
now that you mentioned it, we can just do `len(manifest) + 1` to stay safe regardless of the number of manifests

Reply:
changed the var to hold len(manifestsToApply)